inductiva/tasks/task.py

```diff
@@ -174,13 +175,73 @@ def wait(self, polling_period: int = 5) -> models.TaskStatusCode:
 
             time.sleep(polling_period)
 
-    def kill(self) -> None:
-        """Kill the task.
+    def kill(
+            self,
+            wait_timeout: Optional[Union[float,
+                                         int]] = None) -> Union[bool, None]:
+        """Tries to kill a task.
+        If wait_timeout is None, the function will send the kill
+        request, and return none. If wait_timeout is a positive number,
+        the function will send the kill request and wait (wait_timeout
+        seconds) for the task to be killed. We return True if the task
+        was successfully killed, and False otherwise.
+        Args:
+            wait_timeout: number of seconds to wait for the kill command
+            or none if we don't want to wait.
+        Returns:
+            None if we don't wait, True if the kill command was successful,
+            False if it was not.
 
-        This method issues a request to the API to kill the task. It doesn't
-        block waiting for confirmation if the task was killed.
         """
-        self._api.kill_task(path_params=self._get_path_params())
+        if not isinstance(wait_timeout, (float, int, type(None))) or (
+                wait_timeout is not None and wait_timeout <= 0.0):
+            raise ValueError("Wait timeout must be a positive number or None.")
```

Comment:
we actually have2 types of exceptions here:
- TypeError `if not None and not isinstance(wait_timeout, (int, float))`
- ValueError if `wait_timeout < 0.0`

```suggestion
        if wait_timeout is not None:
            if not isinstance(wait_timeout, (float, int)):
                raise TypeError
            if wait_timeout < 0.0:
                raise ValueError
```